configuration_manager/utils/util.py

```diff
@@ -0,0 +1,33 @@
+import os
+import errno
+
+
+def safe_makedir(target_directory):
+    """Creates a directory.
+
+    Safely creates a directory where there is a race condition i.e.
+    multiple processes try to create the same directory.
+
+    Parameters
+    ----------
+    target_directory : string
+        The Name for the target directory
+
+    Returns
+    -------
+    target_directory: str
+        Target directory
+    """
+    try:
+        os.makedirs(target_directory)
+    except OSError as e:
+        if e.errno == errno.EEXIST and os.path.isdir(target_directory):
+            # The directory already exists.
+            pass
+        else:
+            # Either there exists some file with the same name as the
+            # target directory, or there is some different error.
+            # Just re-raise the error!
+            raise e
+    finally:
+        return target_directory
```

Comment:
Related to the review comment on `directories_manager.py`.
Do we have to think about possible issues if there will be one output directory (already exists) but from different executions of the workflow?

Reply:
We'll handle this in directory_manager.py in future iterations.